staging/src/k8s.io/apiserver/pkg/util/peerproxy/peerproxy_handler.go

```diff
@@ -157,6 +159,7 @@ func (h *peerProxyHandler) WrapHandler(handler http.Handler) http.Handler {
 }
 
 func (h *peerProxyHandler) populateLocalDiscoveryCache() error {
+	clear(h.localDiscoveryResponseCache)
```

Comment:
you are locking the access below

> 	h.discoveryResponseCacheLock.Lock()
	defer h.discoveryResponseCacheLock.Unlock()

is it safe to clean it here? I can see it is used in `localDiscoverySuccessful` too

Reply:
I had removed the `clear(h.localDiscoveryResponseCache)` - shouldnt be happening now.